### PR TITLE
Init Service Indicator counter in `FIPS_service_indicator_update_state`

### PR DESCRIPTION
**Pull request by nebeid**
### Issues:
N/A
Addresses FIPS feedback on PR#342

### Description of changes: 
The description is in the comment in the change.
This PR is based on top of PR#342; only the last commit is the contributed change.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by samuel40791765 — APPROVED** [review]

**Comment by samuel40791765** [comment]
There might be merge conflicts to resolve once https://github.com/awslabs/aws-lc/pull/342 is squashed and merged to main though.

**Review by dkostic — APPROVED** [review]

File: crypto/fipsmodule/service_indicator/service_indicator.c [hunk 1]
```diff
@@ -68,6 +68,13 @@ void FIPS_service_indicator_update_state(void) {
...
       CRYPTO_get_thread_local(AWSLC_THREAD_LOCAL_FIPS_SERVICE_INDICATOR_STATE);
   if (indicator == NULL) {
+    // FIPS 140-3 requires that the module should provide the service indicator for approved
+    // services irrespective of whether the user queries it or not.
+    // Hence, it is not enough to initialise the counter lazily in the external call
+    // |FIPS_service_indicator_before_call|.
+    // Since this function is always called in the approved services,
+    // the counter will be initialised here if needed.
+    FIPS_service_indicator_init_state();
     return;
```
Comment (dkostic) [inline crypto/fipsmodule/service_indicator/service_indicator.c:78]: Shouldn't we continue to lines 80 and 81 instead of returning?

**Review by dkostic — CHANGES_REQUESTED** [review]
Can't dismiss my approval so I had to "request changes" until the comment I left is addressed.

**Comment by samuel40791765** [comment]
The comment in at `crypto/fipsmodule/service_indicator/service_indicator.c#L24` should also be updated.

**Comment by nebeid** [comment]
I ran the new test without the change in `FIPS_service_indicator_update_state()`:
```
[ RUN      ] ServiceIndicatorTest.BasicTest
../crypto/fipsmodule/service_indicator/service_indicator_test.cc:1630: Failure
Expected: (counter_before) != (0), actual: 0 vs 0
[  FAILED  ] ServiceIndicatorTest.BasicTest (0 ms)
```
as expected, which means that the self-test didn't update the counter or anything else.

**Review by samuel40791765 — APPROVED** [review]

**Review by dkostic — APPROVED** [review]

**Review by darylmartin100 — APPROVED** [review]

**Pull request merged**